backend/de.metas.business/src/main/java/de/metas/invoice/invoiceProcessingServiceCompany/InvoiceProcessingServiceCompanyService.java

```diff
@@ -88,9 +91,13 @@ public InvoiceProcessingServiceCompanyService(
 		final InvoiceId invoiceId = request.getInvoiceId();
 		final Amount feeAmountIncludingTax = request.getFeeAmountIncludingTax();
 
-		if (isServiceInvoiceAlreadyGenerated(invoiceId))
+		final ImmutableSet<InvoiceId> invoiceIdsWithGeneratedFees = retainIfServiceInvoiceWasAlreadyGenerated(ImmutableSet.of(invoiceId));
+
+		if (invoiceIdsWithGeneratedFees.size() > 0)
 		{
-			return Optional.empty();
+			throw new AdempiereException("Service Invoice already generated")
```

Comment:
please trl
🗣️ 
- DE `Die Rechnung {0} hat bereits eine zugeordnete Service-Rechnung`
- EN `Invoice {0} already has a service invoice`

where the parameter is the invoice's DocumentNo. that way the user can find it in the remadv-window